### PR TITLE
With

### LINKED ISSUE #120: Create with_example_project()?
Via https://github.com/cynkra/fledge/pull/113#discussion_r708335953

> Combine `create_demo_project()` and `with_project()`. I'm not sure it's a good idea though.

### PR DESCRIPTION
**Pull request by maelle**
Fix  #120

### CODE REVIEW

**Comment by maelle** [comment]
@krlmlr is this what you had in mind? If so I'll finalize this PR and use the function in all examples and tests. :smile:

File: tests/testthat/test-bump_version.R [hunk 1]
```diff
@@ -1,37 +1,14 @@
...
-  )
-  expect_snapshot_file(
-    file.path(repo, "NEWS.md"),
-    compare = compare_file_text
-  )
+  with_demo_project({
+    use_r("bla")
+    gert::git_add("R/bla.R")
+    gert::git_commit("* Add cool bla.")
+    expect_snapshot(bump_version(), variant = as.character(packageVersion("rlang")))
```
Comment (krlmlr) [inline tests/testthat/test-bump_version.R:7]: Same problem with the variant here.

File: R/repo-helper.R [hunk 2]
```diff
@@ -91,3 +91,38 @@ set_usethis_desc <- function(maintainer, email, date) {
...
+#' @examples
+#' with_demo_project({
+#' # Add a new R file.
+#' usethis::use_r("cool-function", open = FALSE)
+#' # Pretend we added useful code inside it.
+#' # Track the new R file with Git.
+#' gert::git_add("R/cool-function.R")
+#' gert::git_commit("- Add cool function.")
+#' # Bump version with fledge.
+#' fledge::bump_version()
```
Comment (krlmlr) [inline R/repo-helper.R:112]: Should we indent this code?

**Review by krlmlr — COMMENTED** [review]
This looks good, thanks! Let's see what the other examples look like after the conversion.

**Pull request closed without merging**